### PR TITLE
[FEATURE] NotificationScheduler

### PR DESCRIPTION
**Pull request by sentientwaffle**
Depends on https://github.com/interledger/five-bells-shared/pull/69

Fixes https://github.com/interledger/five-bells-notary/issues/17

This includes fixes from shared:

* `PersistentKnexModelMixin#save()` was only updating, never inserting.
* `PersistentKnexModelMixin#destroy()` destroyed the parameter, not the callee.

### CODE REVIEW

**Comment by naoitoi** [comment]
What is NotificationScheduler?  Why is it necessary?  Is the requirement written down anywhere?  
I am responsible for maintaining this code base, so I would like to have a good understanding of what goes into it.

**Comment by sentientwaffle** [comment]
`NotificationScheduler` extracts logic related to scheduling notification retries that is common between ledger and notary. Ledger hasn't been refactored yet because it is still using sequelize.

File: src/lib/app.js [hunk 1]
```diff
@@ -46,8 +46,8 @@ module.exports = class App {
       yield knex.knex.migrate.rollback(knex.config)
       yield knex.knex.migrate.latest(knex.config)
     }
-    yield this.notificationWorker.start()
-    yield this.timerWorker.start()
+    this.notificationWorker.start()
+    this.timerWorker.start()
```
Comment (naoitoi) [inline src/lib/app.js]: Why not wait for notificationWorker and timeWorker?  Doesn't the main event loop need these to be ready when it starts?

Comment (sentientwaffle) [inline src/lib/app.js]: Those are synchronous functions.

Comment (naoitoi) [inline src/lib/app.js]: Cool.  LGTM.

**Pull request merged**